### ISSUE TITLE
Documentation: hv.Dimension: positional argument not documented

### ISSUE THREAD

**Issue by wmayner** [ioam/holoviews#2947]
I'm finding myself a bit perplexed by the documentation system used throughout Holoviews. 

For example, when I try to learn how to use `hv.Dimension`, there doesn't seem to be any information on the positional argument (which I now know should be the dimension's name). This issue is similar to #2925.

1. `help(hv.Dimension)` only documents keyword arguments (Params)
2.  `hv.help(hv.Dimension)` only documents keyword arguments
3. [The API docs](http://holoviews.org/Reference_Manual/holoviews.core.html#holoviews.core.Dimension) only document keyword arguments
4. The positional argument is called `spec`, which doesn't indicate that it should be the name of the dimension (I looked at the source and I see why this is the case, but it compounds the above issues)

The only place where `name` is mentioned as an argument is in the [user guide](http://holoviews.org/user_guide/Annotating_Data.html). This is misleading as currently written, because it suggests that the `spec` positional argument is a keyword argument called `name`. It also suggests that `name` is a Param documented by `hv.help(hv.Dimension)`, which isn't the case.

I think non-Param arguments could be documented more clearly throughout.

**Comment by Gordon90s** [comment]
I second that. Took me a few hours to figure it out.

I think it would be great to have examples within the documentation.

**Comment by jlstevens** [comment]
I agree that we should improve the docs here. We should clarify in the user guide that there is one positional argument (called ``spec`` as you say) of which the most common form of ``spec`` is simply the dimension name. Then we can document the less used tuple formats that are the other supported ``spec`` arguments.
